src/Helpers/JWKFetcher.php

```diff
@@ -62,17 +77,43 @@ protected function convertCertToPem(string $cert) : string
         return $output;
     }
 
+    /**
+     * Get a specific kid from a JWKS.
+     *
+     * @param string      $kid     Key ID to get.
+     * @param string|null $jwksUri JWKS URI to use, or fallback on class-level one.
+     *
+     * @return mixed|null
+     */
+    public function getKey(string $kid, string $jwksUri = null)
+    {
+        $jwksUri = $jwksUri ?? $this->jwksUri;
+        $keys    = $this->getKeys( $jwksUri );
+
+        if (! empty( $keys ) && empty( $keys[$kid] )) {
+            $keys = $this->getKeys( $jwksUri, false );
+        }
+
+        return $keys[$kid] ?? null;
+    }
+
     /**
      * Gets an array of keys from the JWKS as kid => x5c.
      *
-     * @param string $jwks_url Full URL to the JWKS.
+     * @param string  $jwks_url  Full URL to the JWKS.
+     * @param boolean $use_cache Set to false to skip cache check; default true to use caching.
      *
      * @return array
      */
-    public function getKeys(string $jwks_url) : array
+    public function getKeys(string $jwks_url = null, bool $use_cache = true) : array
     {
+        $jwks_url = $jwks_url ?? $this->jwksUri;
+        if (empty( $jwks_url )) {
```

Comment:
If both the passed and class level (default) URIs are null, it looks like a configuration error. Wouldn't it be better to throw or fail gracefully, guiding the dev to update the value?

Reply:
I think the graceful failure would happen if there are no keys sent back upstream. There are a lot of different things that *could* go wrong here and all, in this case, would result in an Exception that the kid was not found. That feels OK to me here. 

Some of this comes from the fact that I'm adding this functionality post-major and don't want to introduce any new breaks.